sky/provision/docker_utils.py

```diff
@@ -329,14 +346,19 @@ def initialize(self) -> str:
             'echo "export DEBIAN_FRONTEND=noninteractive" >> ~/.bashrc;',
             run_env='docker')
         # Install dependencies.
-        self._run(
-            'sudo apt-get update; '
+        cmd = (
+            'bash -lc \''
+            'exec 200>/var/tmp/sky_apt.lock; '
+            'flock -x -w 120 200 || exit 1; '
```

Comment:
Don't really see a straightforward path to adding this lock to the _run function.